src/components/EnrollmentApp.tsx

```diff
@@ -1,7 +1,223 @@
+        super(props);
+        this.state = {
+            activeCarePlan: null,
+            editMode: null,
+            error: null
+        };
+    }
+
+    createNewCarePlan(): CarePlan {
+        //@ts-ignore
+        let patient: Patient = this.context.patient;
+        //@ts-ignore
+        let client: Client = this.context.client;
+
+        let planDefinition = getDefaultMessageSchedule();
+
+        let replacements: { [key: string]: number } = {};
+        let preferred_username = getUsername(client);
+        if (preferred_username) {
+            replacements['{userName}'] = preferred_username;
+        }
+
+        const messages: CommunicationRequest[] = planDefinition.createMessageList(patient, replacements);
+        const carePlan = makeCarePlan(planDefinition, patient, messages);
+
+        return carePlan;
+    }
+
+    discontinueExistingCarePlan() {
+        // @ts-ignore
+        let client: Client = this.context.client;
+        //@ts-ignore
+        let existingCarePlan: CarePlan = this.context.carePlan;
+        //@ts-ignore
+        let patient: Patient = this.context.patient;
+
+        let params = new URLSearchParams({
+            "recipient": `Patient/${patient.id}`,
+            "category": IsaccMessageCategory.isaccScheduledMessage.code,
+            "status": "active",
+            "based-on": existingCarePlan.reference,
+            "_sort": "occurrence",
+            "_count": "1000"
+        }).toString();
+
+        client.request(`CommunicationRequest?${params}`).then((bundle: Bundle) => {
+            if (bundle.type !== "searchset") {
+                this.setState({error: `Unexpected bundle type returned: ${bundle.type}`});
+                return null;
+            }
+            if (!bundle.entry) {
+                return null;
+            }
+            let crsToRevoke: CommunicationRequest[] = bundle.entry.map((e: IBundle_Entry) => {
+                if (e.resource.resourceType === 'CommunicationRequest') {
+                    return CommunicationRequest.from(e.resource as ICommunicationRequest);
+                } else {
+                    this.setState({error: `Unexpected resource type returned: ${e.resource.resourceType}`});
+                    return null;
+                }
+            })
+            crsToRevoke.forEach((cr) => {
+                cr.status = "revoked";
+                client.update(cr).then(
+                    //onfulfilled
+                    (value) => console.log(`CommunicationRequest/${cr.id} successfully revoked:`, value),
+                    //onrejected
+                    (rejectionReason) => console.log(`CommunicationRequest/${cr.id} revocation rejected:`, rejectionReason)
+                ).catch(
+                    (errorReason) => console.log(`CommunicationRequest/${cr.id} revocation caused error:`, errorReason)
+                );
+            });
+
+        }).then(() => {
+            existingCarePlan.status = "revoke";
```

Comment:
@hannahburkhardt  I tested this in Smart Health IT launcher.  When I tried clicking on `revoke and create new` to revoke existing CommunicationRequests and CarePlan, it **worked** for **CommunicationRequests** - they were marked as revoked in the status field as expected.  However _it didn't_ for **CarePlan**.  This is the error message I got for CarePlan:  `Error was: Invalid attribute value \"revoke\": Unknown CarePlanStatus code 'revoke'`  I see allowed values for status is slightly different for [CarePlan](https://fhir-ru.github.io/careplan.html) vs [CommunicationRequest](https://fhir-ru.github.io/communicationrequest.html).

Reply:
Are you using R4? I think the value set was CarePlanStatus in R3, but both of the links you shared use RequestStatus for the status attribute in R4

Reply:
ah, okay, I am using R4 and tested in R4.